agent/api/task_test.go

```diff
@@ -194,6 +198,83 @@ func TestDockerHostConfigVolumesFrom(t *testing.T) {
 	}
 }
 
+func TestDockerHostConfigRawConfig(t *testing.T) {
```

Comment:
Can you add a test similar to this showing that non-overridden existing data in Config/HostConfig is properly preserved when merged?